### ISSUE TITLE
Reset deletes all data, not back to when checkpoint was taken

### ISSUE THREAD

**Issue by fatmanmclone90** [jbogard/Respawn#21]
Test class using xunit, with a private static checkpoint for a sql server database:

`private static Checkpoint checkpoint = new Checkpoint
        {
            TablesToIgnore = new[]
        {
        "schema_version"
        },
            SchemasToExclude = new[]
        {
        "dbo"
        }
        };`

In the test I call reset
`[Fact]
        public async Task MyTest()
        {
            checkpoint.Reset("Data Source=localhost;Initial Catalog=Guides;Integrated Security=True;MultipleActiveResultSets=False;Min Pool Size=3;Application Name=Guides.Api");
.
.
.}`

Reset command is deleting all records from the two tables in the 'guides' schema.  Tables have a 1 to many relationship.

Expected only to delete records after checkpoint taken.

Behaviour occurred when using both 0.2.0 and 0.3.0 (preview)

Am I misusing the checkpoint?

**Comment by jbogard** [comment]
Nope, that's the point. It resets all the tables, it does not try to save state. If it did, it'd be much more expensive to run those kinds of tests.

Respawn is for tests where you reset back to a blank slate, except for specific, excluded tables/schemas.

**Comment by JC-Yodiwo** [comment]
You mean we can not respawn an actual database, or the tables will be deleted. Correct?

**Comment by jbogard** [comment]
I don't understand this question, can you rephrase?

**Comment by JC-Yodiwo** [comment]
I have an integration test which add a record in a table (new added record Id = 10), once the integration test finishes, I need to remove the record 10 and keep the already existing records. Is there a way I can do that.

**Comment by jbogard** [comment]
Not with this library, or any others that I'm aware of. Your best bet is probably transactions in a test, roll it back at the end.
